### ISSUE TITLE
getopts: detect option letters handled in case statement but missing from optstring

### ISSUE THREAD

**Issue by phagara** [koalaman/shellcheck#907]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

usage() {
        echo "$0 [-a text] [-b]"
        exit 1
}

while getopts a: opt; do
        case "$opt" in
                a) foo=$OPTARG ;;
                b) alias bar=true ;;
                *) usage ;;
        esac
 done

echo "$foo"

```

#### Here's what shellcheck currently says:

`No issues detected!`

#### Here's what I wanted or expected to see:

Something along the lines of `Attempting to handle option "b" not present in getopts' optstring.`

**Comment by koalaman** [comment]
This feature has been added in 4243c6a. It only works for `getopts` (not `getopt`) in strictly canonical while loops like this one.

Great suggestion. It's high frequency so it'll actually trigger, high precision so it won't be noisy, and high level so you'll be impressed when it's caught ^^

**Comment by brother** [comment]
Incorrect commit link. 
I think this is the correct one https://github.com/koalaman/shellcheck/commit/070a465b64f07600

**Comment by koalaman** [comment]
Oops! Yes, that's the one

**Comment by phagara** [comment]
Awesome, thanks!
